example/game-of-life/game_of_life.cpp

```diff
@@ -0,0 +1,212 @@
+// Conway's game of life
+#include <chrono>
+#include <mcpp/mcpp.h>
+#include <thread>
+
+class Life {
+  private:
+    int width, depth;
+    int** game; // 1 is active
+    bool isRunning;
+    int delay;
+
+    const mcpp::BlockType play_block = mcpp::Blocks::GREEN_CONCRETE;
+    const mcpp::BlockType pause_block = mcpp::Blocks::RED_CONCRETE;
+    const mcpp::BlockType dead_block = mcpp::Blocks::BLACK_CONCRETE;
+    const mcpp::BlockType alive_block = mcpp::Blocks::WHITE_CONCRETE;
+    const mcpp::BlockType speed_block = mcpp::Blocks::YELLOW_CONCRETE;
+    const mcpp::BlockType slow_block = mcpp::Blocks::BLUE_CONCRETE;
+    const mcpp::BlockType random_block = mcpp::Blocks::PINK_CONCRETE;
+    const mcpp::BlockType clear_block = mcpp::Blocks::BARRIER;
```

Comment:
These should probably be all caps according to const convention, and you may want to split them out of the class potentially though that's up to you